### ISSUE TITLE
DataTable: Part 2

### ISSUE THREAD

**Issue by machikoyasuda** [reactioncommerce/catalyst#110]
Things not available in this table:
- [ ] Loading state
- [ ] empty state
- [ ] column sorting (by clicking headers)
- [ ] mobile support
- [x] custom filter controls
- [x] row interactions like clicking the entire row to link to something. Custom cell rendered can be used to make things inside cells clickable, but the entire row is not interactive)
- [ ] Checkbox icon design is not in line with designs. (this should be a global theme prop override to change the checked icon component)
- [ ] Dropdown page selector has a vertical divider. (Best solved in its own PR as hiding the `|` line should be a function of the `Select` component itself)

_Originally posted by @mikemurray in https://github.com/reactioncommerce/catalyst/pull/103#issuecomment-534278296_

**Comment by machikoyasuda** [comment]
Styling bug fix:

The first row should always have a grey background. Maybe need to change the even/odd styles:

<img width="798" alt="Screen Shot 2019-09-24 at 12 11 51 PM" src="https://user-images.githubusercontent.com/3673236/65542730-923df100-dec4-11e9-9934-e35845e7af69.png">

In Zeplin mocks, the first row is grey, then second row is white.

**Comment by janus-reith** [comment]
I really wonder why [react-table](https://github.com/tannerlinsley/react-table) was dropped for this?
There already is a similar looking Table implementation for the Tags, which had pagination with GraphQL implemented and featured custom sorting and searches.
It also was simple to extend to also handle keyword searches and column sorting over graphql.

Before I found the DataTable component here I found the idea very appealing to turn that into some general purpose component to display data from several queries that use the same paradigm.
